uber/models.py

```diff
@@ -1243,9 +1196,10 @@ def _badge_adjustments(self):
 
         self.badge_type = get_real_badge_type(self.badge_type)
 
+        if self.orig_value_of('badge_type') != self.badge_type:
+            self.session.update_badge(self, self.orig_value_of('badge_type'), self.orig_value_of('badge_num'))
+
         if not needs_badge_num(self):
-            if self.orig_value_of('badge_num') and c.SHIFT_CUSTOM_BADGES:
-                self.session.shift_badges(self.orig_value_of('badge_type'), self.orig_value_of('badge_num') + 1, down=True)
```

Comment:
This is basically saying if they used to have a badge number, but they don't need it anymore, then remove their badge number and shift the badges down to fill the gap?

But we don't need to do this anymore because theoretically, the only time they will HAVE a badge number but no longer NEED a badge number is if the `badge_type` changed? And we account for that condition above?